src/containers/DiskSpace/DiskSpaceContainer.tsx

```diff
@@ -1,49 +1,79 @@
-        {date: "2020-01-25T14:43:42.856Z", free: 1343, taken: 1001},
-        {date: "2020-01-26T14:43:42.856Z", free: 200, taken: 20},
-        {date: "2020-01-27T14:43:42.856Z", free: 1234, taken: 1092},
-        {date: "2020-01-28T14:43:42.856Z", free: 926, taken: 925},
-    ];
+export const DiskSpace: React.FC<IDiskSpaceProps> = ({selectedNodeIndex}: IDiskSpaceProps): ReactElement => {
+    const formatDiskData = (diskData: Array<INodeDiskState>): Array<DiskSpaceDataProps> => {
+        const formatedData: Array<DiskSpaceDataProps> = [];
+        diskData.forEach(e => {
+            formatedData.push({
+                date: e.updatedAt,
+                free: Math.round(e.freeSpace / 1000000000),
+                taken: Math.round(e.takenSpace / 1000000000),
+            });
+        });
+        return formatedData;
+    };
+
+    const state = useSelector((state: RootState) => state);
+    const {nodeDiskInfo, nodeList} = state.node;
+    const dispatch = useDispatch();
 
-    const [toolTip, setToolTip] = useState(data[5]);
+    const [toolTip, setToolTip] = useState(formatDiskData(nodeDiskInfo)[0]);
+    const [selectedInterval, setSelectedInterval] = useState("year");
 
     const updateTooltip = (e: any): void => {
         if (e.activePayload) {
-            setToolTip(data[e.activeTooltipIndex]);
+            setToolTip(formatDiskData(nodeDiskInfo)[e.activeTooltipIndex]);
         }
     };
 
+    const tooltipValues = (free: number, taken: number): any => {
+        return [
+            {value: `Free - ${free} GB`, icon: <img src={require("../../assets/icons/polygon.svg")} />},
+            {value: `Taken - ${taken} GB`, icon: <img src={require("../../assets/icons/polygon-dark.svg")} />},
+        ];
+    };
+
+    useEffect(() => {
+        if (nodeList[selectedNodeIndex]) dispatch(getDiskInfo(nodeList[selectedNodeIndex].id, selectedInterval));
+    }, [selectedInterval, selectedNodeIndex, nodeList]);
+
     return (
         <div className="container flex-column vertical-margin">
             <div className="upper">
                 <label>disk space</label>
             </div>
-
-            <ChartHeader
-                date={toolTip.date}
-                values={[
-                    {value: `Free - ${toolTip.free} GB`, icon: <img src={require("../../assets/icons/polygon.svg")} />},
-                    {
-                        value: `Taken - ${toolTip.taken} GB`,
-                        icon: <img src={require("../../assets/icons/polygon-dark.svg")} />,
-                    },
-                ]}
-            />
-
-            <DiskSpaceChart data={data} onMouseMove={updateTooltip} />
+            {toolTip ? (
+                <ChartHeader
+                    onIntervalClick={e => setSelectedInterval(e)}
+                    date={toolTip.date}
+                    values={tooltipValues(toolTip.free, toolTip.taken)}
+                />
+            ) : nodeDiskInfo && nodeDiskInfo[0] ? (
+                <ChartHeader
```

Comment:
Umm, no, this is not the case. We use tooltip value always. If you remember how the mocked that is (or you can see currently on other graphs) this header represents the data that is currently hovered.